pkg/controller/machine.go

```diff
@@ -362,7 +388,18 @@ func (c *controller) machineCreate(machine *v1alpha1.Machine, driver driver.Driv
 		c.updateMachineStatus(machine, lastOperation, currentStatus)
 		return err
 	}
-	actualProviderID, nodeName, err := driver.Create()
+	// Before actually creating the machine, we should once check and adopt if the virtual machine already exists.
+	VMList, err := driver.GetVMs("")
+	for providerID, machineName := range VMList {
+		if machineName == machine.Name {
+			klog.V(2).Infof("Adopted an existing VM %s for machine object %s.", providerID, machineName)
+			actualProviderID = providerID
+		}
+	}
+	if actualProviderID == "" {
+		actualProviderID, nodeName, err = driver.Create()
```

Comment:
In case VM already exists, we don't get to this logic, and we end up updating `clone.Status.Node = ""` since `nodeName = ""`.
Is this supposed to be taken care of in the next `updateMachineState()` call? It could take 10 minutes according to the deferred `enqueueMachineAfter` call, right.
Same goes for `syncMachineNodeTemplates` call.

Reply:
Yes, this is a very good point. Basically, if the nodeName is mising then both node-label and `Status.Node` will be set to "" unfortunately, because we dont have a way to get the nodeName at that time. We reconcile the machine-objects on events[add/update] from node-object as well, it means both label and `Status.Node` will adopt the node-name on the next reconcilliation of the machine-object, which doesn't have to be 10mins but till the time node really joins.

Reply:
As discussed over call, we can address these concerns separately.